internal/component/prometheus/operator/configgen/config_gen_scrapeconfig.go

```diff
@@ -0,0 +1,135 @@
+		}
+	}
+	return cfg, errors
+}
+
+func (cg *ConfigGenerator) generateStaticScrapeConfigConfig(m *promopv1alpha1.ScrapeConfig, sc promopv1alpha1.StaticConfig, i int) (cfg *config.ScrapeConfig, err error) {
+	relabels := cg.initRelabelings()
+	metricRelabels := relabeler{}
+	cfg, err = cg.commonScrapeConfigConfig(m, i, &relabels, &metricRelabels)
+	cfg.JobName = fmt.Sprintf("scrapeConfig/%s/%s/static/%d", m.Namespace, m.Name, i)
+	if err != nil {
+		return nil, err
+	}
+	targets := []model.LabelSet{}
+	for _, target := range sc.Targets {
+		targets = append(targets, model.LabelSet{
+			model.AddressLabel: model.LabelValue(target),
+		})
+	}
+	labels := model.LabelSet{}
+	// promote "__address__" to "instance" label.
+	relabels.add(&prelabel.Config{
+		SourceLabels: model.LabelNames{model.AddressLabel},
+		TargetLabel:  model.InstanceLabel,
+	})
+	for k, v := range sc.Labels {
+		labels[model.LabelName(k)] = model.LabelValue(v)
+	}
+	config := discovery.StaticConfig{
+		&targetgroup.Group{
+			Targets: targets,
+			Labels:  labels,
+			Source:  cfg.JobName,
+		},
+	}
+	cfg.ServiceDiscoveryConfigs = append(cfg.ServiceDiscoveryConfigs, config)
+	cfg.RelabelConfigs = relabels.configs
+	cfg.MetricRelabelConfigs = metricRelabels.configs
+	return cfg, cfg.Validate(cg.ScrapeOptions.GlobalConfig())
+}
+
+func (cg *ConfigGenerator) commonScrapeConfigConfig(m *promopv1alpha1.ScrapeConfig, i int, relabels *relabeler, metricRelabels *relabeler) (cfg *config.ScrapeConfig, err error) {
+	cfg = cg.generateDefaultScrapeConfig()
+	if m.Spec.HonorLabels != nil {
+		cfg.HonorLabels = *m.Spec.HonorLabels
+	}
+	if m.Spec.HonorTimestamps != nil {
+		cfg.HonorTimestamps = *m.Spec.HonorTimestamps
+	}
+	if m.Spec.ScrapeInterval != nil {
+		if cfg.ScrapeInterval, err = model.ParseDuration(string(*m.Spec.ScrapeInterval)); err != nil {
+			return nil, fmt.Errorf("parsing interval from scrapeConfig: %w", err)
+		}
+	}
+	if m.Spec.ScrapeTimeout != nil {
+		if cfg.ScrapeTimeout, err = model.ParseDuration(string(*m.Spec.ScrapeTimeout)); err != nil {
+			return nil, fmt.Errorf("parsing timeout from scrapeConfig: %w", err)
+		}
+	}
+	if m.Spec.MetricsPath != nil {
+		cfg.MetricsPath = *m.Spec.MetricsPath
+	}
+	if m.Spec.Params != nil {
+		cfg.Params = m.Spec.Params
+	}
+	if m.Spec.Scheme != nil {
+		cfg.Scheme = *m.Spec.Scheme
+	}
+	if m.Spec.TLSConfig != nil {
+		if cfg.HTTPClientConfig.TLSConfig, err = cg.generateSafeTLS(*m.Spec.TLSConfig, m.Namespace); err != nil {
+			return nil, err
+		}
+	}
+	if m.Spec.BasicAuth != nil {
+		cfg.HTTPClientConfig.BasicAuth, err = cg.generateBasicAuth(*m.Spec.BasicAuth, m.Namespace)
+		if err != nil {
+			return nil, err
+		}
+	}
+	if m.Spec.Authorization != nil {
+		cfg.HTTPClientConfig.Authorization, err = cg.generateAuthorization(*m.Spec.Authorization, m.Namespace)
+		if err != nil {
+			return nil, err
+		}
+	}
+	relabels.add(&relabel.Config{
+		Replacement: m.Namespace,
+		TargetLabel: "__meta_kubernetes_scrapeconfig_namespace",
```

Comment:
Is this label documented?

Reply:
@ptodev Can you please point me at the right place for documenting such labels? And maybe a prior art example from another component?

Reply:
@ptodev I addressed this in https://github.com/alex-berger/alloy/commit/f1dea86b042c3a649a77626eebaf91b891ec37c0, let me know if there is a better place for it.